src/lib/components/shared/vercel_hover.svelte

```diff
@@ -0,0 +1,72 @@
+<script lang="ts">
+    export let glider_container_class = "";
+    export let active_element_class = "";
+    export let direction: "horizontal" | "vertical";
+
+    /** vercel effect */
+    let hover_glider_element: HTMLElement,
+        glider_container_element: HTMLElement,
+        GLIDER_TRANSITION_DURATION = 200,
+        is_hovered = false,
+        mouse_leave_timeout: NodeJS.Timer | null = null;
+
+    const handle_mouseenter = (event: MouseEvent) => {
+        const target = event.target as HTMLElement;
+        const target_computed_style = getComputedStyle(target);
+
+        // To make sure our operations are proper we need to make sure that the `position` is set to relative
+        glider_container_element.style.position = "relative";
+
+        // Do some magic here to get the target's height and width
+        // Don't change the position of this code.
+        // It will cause animation jank
+        hover_glider_element.style.height = target_computed_style.height;
+        hover_glider_element.style.width = target_computed_style.width;
+
+        // We need to make sure that zIndex is not auto
+        const target_zindex = parseInt(target_computed_style.zIndex);
+        hover_glider_element.style.zIndex = String(target_zindex ? target_zindex - 1 : -1);
```

Comment:
Also why these codes?
add `-z-10` class? we're trying to reduce js codes, right?

Reply:
Bad idea

Reply:
Bad idea